oauthlib/oauth2/rfc6749/tokens.py

```diff
@@ -331,17 +345,13 @@ def create_token(self, request, refresh_token=False, save_token=False):
         return self.request_validator.get_jwt_bearer_token(None, None, request)
 
     def validate_request(self, request):
-        token = None
-        if 'Authorization' in request.headers:
-            token = request.headers.get('Authorization')[7:]
-        else:
-            token = request.access_token
+        token = get_token_from_header(request)
         return self.request_validator.validate_jwt_bearer_token(
             token, request.scopes, request)
 
     def estimate_type(self, request):
-        token = request.headers.get('Authorization', '')[7:]
-        if token.startswith('ey') and token.count('.') in (2, 4):
+        token_type, token, *_ = request.headers.get('Authorization', '').split(' ')
```

Comment:
That syntax won't fly on Python 2.7...

And should we use `.split()` instead of `.split(' ')` to allow for multispaces then?

Reply:
Also, this can now raise a `ValueError`, which I'm not sure is desired (though haven't checked in detail).

Reply:
dang you're right, too focused on python3. For the multispace I don't know, doesn't `1*SP` stays for 1 space?

Reply:
I know the feeling :)

Re `1*SP` is one or more spaces... unnatural indeed.

Reply:
aaaaa gotcha, fixing also that then! :)

Reply:
fixed both @skion